common/quotas/clocked_rate_limiter_test.go

```diff
@@ -151,3 +151,65 @@ func TestClockedRateLimiter_Wait_DeadlineWouldExceed(t *testing.T) {
 	t.Cleanup(cancel)
 	assert.ErrorIs(t, rl.Wait(ctx), quotas.ErrRateLimiterReservationWouldExceedContextDeadline)
 }
+
+// test that reservations for 1 token ARE unblocked by RecycleToken
+func TestClockedRateLimiter_Wait_Recycle(t *testing.T) {
+	t.Skip("flaky test")
+	t.Parallel()
+	ts := clock.NewEventTimeSource()
+	rl := quotas.NewClockedRateLimiter(rate.NewLimiter(1, 1), ts)
+	ctx := context.Background()
+
+	// take first token
+	assert.NoError(t, rl.Wait(ctx))
+
+	// wait for next token and report when success
+	asserted := false
```

Comment:
should be an atomic.Bool if you writing from one goroutine and reading from another. or use a channel.